x/ecocredit/basket/msg_put.go

```diff
@@ -39,12 +39,8 @@ func (m MsgPut) ValidateBasic() error {
 
 	if len(m.Credits) > 0 {
 		for _, credit := range m.Credits {
-			if len(credit.BatchDenom) == 0 {
-				return sdkerrors.ErrInvalidRequest.Wrap("credit batch denom cannot be empty")
-			}
-
 			if err := core.ValidateBatchDenom(credit.BatchDenom); err != nil {
-				return sdkerrors.ErrInvalidRequest.Wrap(err.Error())
+				return err
```

Comment:
Here's an example of where we should wrap the error with `sdkerrors.ErrInvalidRequest`.

The error returned from the utility function should not already be wrapped with the `sdkerrors.ErrInvalidRequest` as the utility function may be used outside the context of a request and the parse error might b the preferred error to return in the utility function, which is then wrapped here with `sdkerrors.ErrInvalidRequest`.